src/librustc/mir/interpret/value.rs

```diff
@@ -66,6 +66,32 @@ impl<'tcx> ConstValue<'tcx> {
             ConstValue::Scalar(val) => Some(val),
         }
     }
+
+    pub fn try_to_bits(&self, size: Size) -> Option<u128> {
+        self.try_to_scalar()?.to_bits(size).ok()
+    }
+
+    pub fn try_to_bits_for_ty(
+        &self,
+        tcx: TyCtxt<'tcx>,
+        param_env: ParamEnv<'tcx>,
+        ty: Ty<'tcx>,
+    ) -> Option<u128> {
+        let size = tcx.layout_of(param_env.with_reveal_all().and(ty)).ok()?.size;
```

Comment:
Are there no caveats here? I remember @eddyb saying that "reveal all" is often not the right choice.

Reply:
Lately I've had to say almost the exact opposite (because some misunderstanding around `Reveal::All`'s ability to cope with polymorphic situations: modulo bugs, it should always work), so I'm not sure what the context was for what you're thinking of.

Note that `layout_of`, IIRC, uses `with_reveal_all` internally, so if this does anything, it avoids extra queries.

Reply:
I used the same logic as `Const::try_eval_bits`.